src/state/profile/hooks.ts

```diff
@@ -18,47 +17,32 @@ export const useFetchProfile = () => {
   }, [account, dispatch])
 }
 
-export const useProfileForAddress = (address: string) => {
-  const [profileState, setProfileState] = useState({ profile: null, isFetching: false })
-  useEffect(() => {
-    const fetchProfileForAddress = async () => {
-      setProfileState({ profile: null, isFetching: true })
-      try {
-        const profile = await getProfile(address)
-        setProfileState({ profile, isFetching: false })
-      } catch (error) {
-        console.error(`Failed to fetch profile for address ${address}`, error)
-        setProfileState({ profile: null, isFetching: false })
-      }
-    }
-    if (!profileState.isFetching && !profileState.profile) {
-      fetchProfileForAddress()
-    }
-  }, [profileState, address])
-  return profileState
-}
-
 export const useProfile = () => {
   const { isInitialized, isLoading, data, hasRegistered }: ProfileState = useSelector((state: State) => state.profile)
   return { profile: data, hasProfile: isInitialized && hasRegistered, isInitialized, isLoading }
 }
 
 export const useGetProfileAvatar = (account: string) => {
   const profileAvatar = useSelector((state: State) => state.profile.profileAvatars[account])
-  const { username, nft } = profileAvatar || {}
+  const { username, nft, hasRegistered, usernameFetchStatus, avatarFetchStatus } = profileAvatar || {}
   const dispatch = useAppDispatch()
 
   useEffect(() => {
     const address = isAddress(account)
 
-    if (!nft && address) {
+    if (!nft && avatarFetchStatus !== ProfileAvatarFetchStatus.FETCHED && address) {
       dispatch(fetchProfileAvatar(account))
     }
 
-    if (!username && nft && address) {
-      dispatch(fetchProfileUsername(account))
+    if (
+      !username &&
+      avatarFetchStatus === ProfileAvatarFetchStatus.FETCHED &&
```

Comment:
this check is here to prevent fetching username before we figured out if they're registered or not (checked during avatar fetching)